### PR TITLE
Add DeviceController deleteFromFabricTableOnShutdown option

### PR DESCRIPTION
**Pull request by ksperling-apple**
This is analogous to removeFromFabricTableOnShutdown but does an actual Delete(), i.e. affects storage, rather than just a Forget().

Use this to simplify controller shutdown on Darwin.

### CODE REVIEW

File: src/controller/CHIPDeviceController.cpp [hunk 1]
```diff
@@ -400,13 +401,13 @@ void DeviceController::Shutdown()
         // existing sessions too?
         mSystemState->SessionMgr()->ExpireAllSessionsForFabric(mFabricIndex);
 
-        if (mRemoveFromFabricTableOnShutdown)
+        if (mDeleteFromFabricTableOnShutdown)
         {
-            FabricTable * fabricTable = mSystemState->Fabrics();
-            if (fabricTable != nullptr)
```
Comment (ksperling-apple) [inline src/controller/CHIPDeviceController.cpp:406]: We know the SystemState is valid here, so Fabrics() never returns null.

File: src/controller/CHIPDeviceController.h [hunk 2]
```diff
@@ -393,7 +405,8 @@ class DLL_EXPORT DeviceController : public AbstractDnssdDiscoveryController
 
     FabricIndex mFabricIndex = kUndefinedFabricIndex;
 
-    bool mRemoveFromFabricTableOnShutdown = true;
+    bool mRemoveFromFabricTableOnShutdown;
```
Comment (ksperling-apple) [inline src/controller/CHIPDeviceController.h:408]: These get initialized in `Init()`, so no need to repeat any defaults here.

File: src/controller/CHIPDeviceController.cpp (hunk 1 above)
Comment (andy31415) [reply src/controller/CHIPDeviceController.cpp:406]: Should this be in a code comment then?

File: src/controller/CHIPDeviceController.h (hunk 2 above)
Comment (andy31415) [reply src/controller/CHIPDeviceController.h:408]: Seems like good practice though .... please keep initializing them.

**Review by bzbarsky-apple — APPROVED** [review]
Approved, with @andy31415's comments addressed.

File: src/controller/CHIPDeviceController.h (hunk 2 above)
Comment (ksperling-apple) [reply src/controller/CHIPDeviceController.h:408]: I don’t particularly like having the defaults repeated in 3 places, but Ok.

File: src/controller/CHIPDeviceController.cpp (hunk 1 above)
Comment (ksperling-apple) [reply src/controller/CHIPDeviceController.cpp:406]: Added a comment

**Review by andy31415 — APPROVED** [review]

**Pull request merged**